src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkSectionUnexpectedKeywordArgsInspection.kt

```diff
@@ -27,6 +28,12 @@ class SmkSectionUnexpectedKeywordArgsInspection : SnakemakeInspection() {
             }
         }
 
+        override fun visitSmkWorkflowArgsSection(st: SmkWorkflowArgsSection) {
+            if (st.sectionKeyword in SECTIONS_WHERE_KEYWORD_ARGS_PROHIBITED) {
```

Comment:
You check workflow top-level section name in list of rule/checkpoint section names. Here should be 2 different lists, one for rule/checkpoint set, the other for workflow top level sections. E.g. for 'container' we have SnakemakeNames.WORKFLOW_CONTAINER_KEYWORD and SnakemakeNames.CONTAINER_KEYWORD.